src/renderer/entities/network/lib/networkService.ts

```diff
@@ -1,388 +1,115 @@
 import { ApiPromise, ScProvider, WsProvider } from '@polkadot/api';
-import * as Sc from '@substrate/connect';
 import { ProviderInterface } from '@polkadot/rpc-provider/types';
-import keyBy from 'lodash/keyBy';
-import { useRef, useState } from 'react';
+import * as Sc from '@substrate/connect';
+import { noop } from '@polkadot/util';
 
-import { storage } from '@shared/api/storage';
-import { useSubscription } from '../../../services/subscription/subscriptionService';
-import { useChainSpec } from './chainSpecService';
-import { chainsService } from './chainsService';
-import { useMetadata } from './metadataService';
-import { AUTO_BALANCE_TIMEOUT, MAX_ATTEMPTS, PROGRESSION_BASE } from './common/constants';
-import { ConnectionsMap, ConnectProps, INetworkService, RpcValidation } from './common/types';
+import { UniversalProvider } from './provider/UniversalProvider';
+import { ChainId } from '@shared/core';
 import { createCachedProvider } from './provider/CachedProvider';
-import { ConnectionType, ConnectionStatus } from '@shared/core';
-import type { ChainId, Chain, Connection, RpcNode } from '@shared/core';
-
-export const useNetwork = (): INetworkService => {
-  const chains = useRef<Record<ChainId, Chain>>({});
-  const [connections, setConnections] = useState<ConnectionsMap>({});
-
-  const { subscribe, unsubscribe } = useSubscription<ChainId>();
-  const { getKnownChain, getLightClientChains } = useChainSpec();
-  const { subscribeMetadata, getMetadata } = useMetadata();
-
-  const connectionStorage = storage.connectTo('connections');
+import { Metadata } from './common/types';
+import { chainSpecService } from './chainSpecService';
+
+export const networkService = {
+  createProvider,
+  createApi,
+  connect,
+  disconnect,
+};
 
-  if (!connectionStorage) {
-    throw new Error('=== 🔴 Connections storage in not defined 🔴 ===');
+async function createApi(
+  provider: ProviderInterface,
+  onConnected?: () => void,
+  onDisconnected?: () => void,
+  onError?: () => void,
+) {
+  const api = await ApiPromise.create({ provider, throwOnConnect: true, throwOnUnknown: true });
+
+  api.on('connected', onConnected || noop);
+  api.on('disconnected', onDisconnected || noop);
+  api.on('error', onError || noop);
+
+  return api;
+}
+
+function createProvider(
+  chainId: ChainId,
+  nodes: string[],
+  getMetadata: (chainId: ChainId) => Promise<Metadata | undefined>,
+  onConnected?: (value?: any) => void,
+  onDisconnected?: (value?: any) => void,
+  onError?: (value?: any) => void,
+): UniversalProvider {
+  let provider;
+
+  try {
+    const scProvider = createSubstrateProvider(chainId, getMetadata);
```

Comment:
Does SC provider do anything hevy until its connected? If it does, we probably want to delay its initialization until we really need to connect to SC, to avoid extra load on the app start